### ISSUE TITLE
Incorperate Levans' guide into official documentation

### ISSUE THREAD

**Issue by aaronraimist** [matrix-org/synapse#14395]
The author of https://levans.fr/shrink-synapse-database.html said the blog post may disappear in the next few months https://matrix.to/#/!ehXvUhWNASUkSLvAGP:matrix.org/$gb32QeAGPqopUPI2zlQTC7MHV8PAJYsrMKTmqPIh-LM?via=matrix.org&via=libera.chat&via=matrix.breakpointingbad.com

Worst case scenario people can continue to send links to the Wayback Machine archive https://web.archive.org/web/20221107150752/https://levans.fr/shrink-synapse-database.html but it would be nice to add the guide to `admin_faq.md` or somewhere else in the official documentation

**Comment by DMRobertson** [comment]
TL;DR:

1. manually purge forgotten rooms, in lieu of
2. [run the state compressor](https://github.com/matrix-org/rust-synapse-compress-state) on sufficiently large rooms
3. reindex and vacuum full

Synapse should do (1) itself ( https://github.com/matrix-org/synapse/issues/4720); we'd like it to do (2) itself too. (3) is best left to the discretion of the database administrator: both REINDEX and VACUUM FULL require exclusive locks on tables. (There is `REINDEX CONCURRENTLY`, but it comes [with several caveats](https://www.postgresql.org/docs/current/sql-reindex.html#SQL-REINDEX-CONCURRENTLY).)

**Comment by clokep** [comment]
> (2) [run the state compressor](https://github.com/matrix-org/rust-synapse-compress-state) on sufficiently large rooms
>
> we'd like it to do (2) itself too.

See #10857
